Sources/SwiftGenKit/Parsers/Parser.swift

```diff
@@ -8,13 +8,28 @@ import Foundation
 import PathKit
 
 public protocol Parser {
+  /// Parser initializer. Please use the `createWith(options:warningHandler:)` factory method instead,
+  /// it'll ensure the safety of the provided options.
+  ///
+  /// - Parameter options: Dictionary of options, used to customize processing.
+  /// - Parameter warningHandler: Callback for logging issues.
   init(options: [String: Any], warningHandler: MessageHandler?) throws
 
-  // regex for the default filter
+  /// Regex for the default filter
   static var defaultFilter: String { get }
 
-  // Parsing and context generation
+  /// List of supported options
+  static var allOptions: ParserOptionList { get }
```

Comment:
Should we provide a default value of `[]` (no option) for this one?

Reply:
Dunno. We don't for filter (although that'd be nonsensical, and without the default, it's clear in each parser what it can & can't do.

Reply:
Although less code is always nice.